decidim-core/app/packs/src/decidim/data_consent/consent_manager.test.js

```diff
@@ -0,0 +1,280 @@
+global.window = Object.create(window);
+global.window.location = {
+  protocol: "https:",
+  hostname: "decidim.dev"
+};
+
+describe("ConsentManager", () => {
+  const dialogContent = `
+    <div id="cc-dialog-wrapper" class="flex-center" role="region">
+      <div class="cc-dialog padding-vertical-1">
+        <div class="row expanded">
+          <div class="columns medium-12 large-8">
+            <div class="cc-content">
+              <div class="h5">Information about the cookies used on the website</div>
+              <div>
+                We use cookies on our website to improve the performance and content of the site. The cookies enable us to provide a more individual user experience and social media channels.
+              </div>
+            </div>
+          </div>
+          <div class="columns medium-12 large-4">
+            <div class="cc-button-wrapper flex-center">
+              <button id="cc-dialog-accept" class="button">
+                Accept all
+              </button>
+              <button id="cc-dialog-reject" class="button hollow">
+                Accept only essential
+              </button>
+              <button id="cc-dialog-settings" class="button clear" data-open="cc-modal" aria-controls="cc-modal" aria-haspopup="dialog" tabindex="0">
+                Settings
+              </button>
+            </div>
+          </div>
+        </div>
+      </div>
+    </div>
+  `;
+  const modalContent = `
+    <div class="reveal cc-modal" id="cc-modal" role="dialog" data-close-on-click="false" data-close-on-esc="false" aria-modal="true" aria-hidden="true">
+      <div class="reveal__header">
+        <h3 class="reveal__title">Cookie settings</h3>
+        <p>
+          We use cookies to ensure the basic functionalities of the website and to enhance your online experience. You can choose for each category to opt-in/out whenever you want.
+        </p>
+      </div>
+
+      <div class="cc-categories">
+        <div class="category-wrapper margin-vertical-1" data-id="essential">
+          <div class="category-row flex-center">
+            <button class="cc-title padding-left-3">
+              <span class="h5 cc-category-title">
+                <strong>Essential</strong>
+              </span>
+            </button>
+            <div class="cc-switch">
+              <input class="switch-input" checked="checked" id="cc-essential" type="checkbox" name="essential" disabled="">
+              <label class="switch-paddle" for="cc-essential">
+                <span class="show-for-sr">Toggle Essential</span>
+              </label>
+            </div>
+          </div>
+          <div class="cc-description hide">
+            <div class="description-text">
+              <p>These cookies are essential for the proper functioning of my website. Without these cookies, the website would not work properly.</p>
+            </div>
+            <div class="cookie-details-wrapper">
```

Comment:
I don't know how exhaustive we want to be with this renaming, but the only thing I could find is just these CSS classes

```console
decidim-core/app/packs/stylesheets/decidim/modules/_data-consent.scss
76:    .cookie-details-wrapper{

decidim-core/app/cells/decidim/data_consent/category.erb
29:      <div class="cookie-details-wrapper">

decidim-core/app/packs/src/decidim/data_consent/consent_manager.test.js
86:            <div class="cookie-details-wrapper">
```

Reply:
I was actually thinking about this as well but I thought it would be user-facing.

But anyways this is for developers and we both thought the same thing, so I went ahead and did this.

I also renamed the `cc-` prefixes in the data consent related elements to `dc-`.

Hopefully not to confuse with Douglas Commercial.

![DC-3](https://upload.wikimedia.org/wikipedia/commons/thumb/7/79/OH-LCH_DC-3.JPG/1280px-OH-LCH_DC-3.JPG)